src/infinity-menu.js

```diff
@@ -261,6 +261,7 @@ export default class InfinityMenu extends React.Component {
 
 		/*header component*/
 		const headerProps = {
+			headerMetadata: this.props.headerMetadata,
```

Comment:
Yeah, that's true, to me though this gives the flexibility of using an already existing component, and the ability to override existing props which I think is valuable, but agree as long as it's clear it's fine.